ledger/notary-plugin/src/main/kotlin/net/corda/v5/ledger/notary/plugin/api/PluggableNotaryClientFlowProvider.kt

```diff
@@ -0,0 +1,17 @@
+package net.corda.v5.ledger.notary.plugin.api
+
+import net.corda.v5.ledger.common.Party
+import net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction
+
+/**
+ * This interface is used to instantiate the client-side logic (flow) of the plugin.
+ *
+ * Implementers of this interface must have the [PluggableNotaryType] annotation,
+ * otherwise an exception will be thrown on loading.
+ *
+ * The client-side of the plugin must have the same constructor parameters as the
+ * [create] function (notary, stx).
+ */
+interface PluggableNotaryClientFlowProvider {
+    fun create(notary: Party, stx: UtxoSignedTransaction): PluggableNotaryClientFlow
```

Comment:
I understand your concern but I'm not entirely sure what would be the best solution here. Overloading is something I'd like to avoid. Maybe we could use a type parameter and make this interface generic, so it can be extended in the future?